internal/db/schema/migrations/oss/postgres/50/01_static_credential.up.sql

```diff
@@ -0,0 +1,113 @@
+begin;
+
+  create table credential_static_store (
+    public_id wt_public_id primary key,
+    scope_id wt_scope_id not null
+      constraint iam_scope_fkey
+        references iam_scope (public_id)
+        on delete cascade
+        on update cascade,
+    name wt_name,
+    description wt_description,
+    create_time wt_timestamp,
+    update_time wt_timestamp,
+    version wt_version,
+    constraint credential_store_fkey
+      foreign key (scope_id, public_id)
+      references credential_store (scope_id, public_id)
+      on delete cascade
+      on update cascade,
+    constraint credential_static_store_scope_id_name_uq
+      unique(scope_id, name)
+  );
+  comment on table credential_static_store is
+    'credential_static_store is a table where each row is a resource that represents a static credential store. '
+    'It is a credential_store subtype and an aggregate root.';
+
+  create trigger update_version_column after update on credential_static_store
+    for each row execute procedure update_version_column();
+
+  create trigger update_time_column before update on credential_static_store
+    for each row execute procedure update_time_column();
+
+  create trigger default_create_time_column before insert on credential_static_store
+    for each row execute procedure default_create_time();
+
+  create trigger immutable_columns before update on credential_static_store
+    for each row execute procedure immutable_columns('public_id', 'scope_id','create_time');
+
+  create trigger insert_credential_store_subtype before insert on credential_static_store
+    for each row execute procedure insert_credential_store_subtype();
+
+  create trigger delete_credential_store_subtype after delete on credential_static_store
+    for each row execute procedure delete_credential_store_subtype();
+
+  create table credential_static_username_password_credential (
+    public_id wt_public_id primary key,
+    store_id wt_public_id not null
+      constraint credential_static_store_fkey
+        references credential_static_store (public_id)
+        on delete cascade
+        on update cascade,
+    name wt_name,
+    description wt_description,
+    create_time wt_timestamp,
+    update_time wt_timestamp,
+    version wt_version,
+
+    username text not null
+      constraint username_must_not_be_empty
+        check(length(trim(username)) > 0),
+    password_encrypted bytea not null
+      constraint password_encrypted_must_not_be_empty
+        check(length(password_encrypted) > 0),
+    password_hmac bytea not null
+      constraint password_hmac_must_not_be_empty
+        check(length(password_hmac) > 0),
+    dek_id text not null
```

Comment:
Any reason going to `dek_id` over `key_id` as we do for other tables? I think we still need to have the struct in go be `KeyID` for the decrypt. Maybe not.  Either way this change might be good since it clearly states what specific key this this, just curious on thoughts?

Reply:
I'm sort of with Louis on this... can we be consistent and use key_id?